src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -30,57 +29,106 @@
+     * This method is called by spark-swagger to register endpoints and generate the docs.
+     * @param restApi The object to which to attach the documentation.
      */
-    private void registerRoutes() {
-        LOG.info("Registering routes for {}", ROOT_ROUTE);
-        // Options response for CORS
-        options(ROOT_ROUTE, (q, s) -> "");
-        // Get multiple entities.
-        get(ROOT_ROUTE, this::getMany, JsonUtils::toJson);
-        // Get one entity.
-        get(ROOT_ROUTE + ID_PARAM, this::getOne, JsonUtils::toJson);
-        // Create entity request
-        post(ROOT_ROUTE, this::createOrUpdate, JsonUtils::toJson);
-        // Update entity request
-        put(ROOT_ROUTE + ID_PARAM, this::createOrUpdate, JsonUtils::toJson);
-        // Delete entity request
-        delete(ROOT_ROUTE + ID_PARAM, this::deleteOne, JsonUtils::toJson);
+    @Override
+    public void bind(final SparkSwagger restApi) {
+        LOG.info("Registering routes and enabling docs for {}", ROOT_ROUTE);
+
+        restApi.endpoint(endpointPath(ROOT_ROUTE)
+                .withDescription("Interface for querying and managing '" + classToLowercase + "' entities."), (q, a) -> LOG.info("Received request for '{}' Rest API", classToLowercase))
+
+
+                // If using lambdas with the GET method, a bug in the documentation framework
+                // requires you to write path(<entire_route>).
+                // If you use new GsonRoute() {...} with the GET method, you only need to write path(<relative_to_endpointPath>).
+                // Other HTTP methods are not affected by this bug.
+
+                // Get multiple entities.
+                // get(ROOT_ROUTE, this::getMany, JsonUtils::toJson);
+                .get(path(ROOT_ROUTE)
+                        .withDescription("Gets a list of all '" + classToLowercase + "' entities.")
+                        .withResponseAsCollection(clazz),
+                        this::getMany, JsonUtils::toJson
+                )
+
+                // Get one entity.
+                // get(ROOT_ROUTE + ID_PARAM, this::getOne, JsonUtils::toJson);
+                .get(path(ROOT_ROUTE + ID_PARAM)
+                        .withDescription("Returns a '" + classToLowercase + "' entity with the specified id, or 404 if not found.")
+                        .withPathParam().withName("id").withDescription("The id of the entity to search.").and()
+                        // .withResponses(...) // FIXME: not implemented (requires source change).
+                        .withResponseType(clazz),
+                        this::getOne, JsonUtils::toJson
+                )
+
+                // Options response for CORS
+                // options(ROOT_ROUTE, (q, s) -> "");
+                .options(path(""), (req, res) -> "")
+
+                // Create entity request
+                // post(ROOT_ROUTE, this::createOrUpdate, JsonUtils::toJson);
+                .post(path("")
```

Comment:
Why is the path an empty string literal here, but `ROOT_ROUTE` in other places? I'm a little confused, but perhaps just a comment is needed to explain why?

Reply:
See comment at L63.